### ISSUE TITLE
Problems in training the model

### ISSUE THREAD

**Issue by VictorLi2017** [OpenNMT/OpenNMT-py#158]
Hello there, I tried to train the model following the instruction on OpenNMT github on a GPU machine, but got problems. After prepossing the data, I enter the command:
`python train.py -data data/demo.train.pt -save_model demo-model`

I got errors:
Namespace(attention_type='general', batch_size=64, brnn=False, brnn_merge='concat', context_gate=None, copy_attn=False, coverage_attn=False, curriculum=False, data='data/demo.train.pt', decay_method='', decoder_layer='rnn', dropout=0.3, encoder_layer='rnn', encoder_type='text', epochs=13, experiment_name='', extra_shuffle=False, feat_merge='concat', feat_vec_exponent=0.7, feat_vec_size=20, gpus=[], input_feed=1, lambda_coverage=1, layers=2, learning_rate=1.0, learning_rate_decay=0.5, log_interval=50, log_server='', max_generator_batches=32, max_grad_norm=5, optim='sgd', param_init=0.1, position_encoding=False, pre_word_vecs_dec=None, pre_word_vecs_enc=None, rnn_size=500, rnn_type='LSTM', save_model='demo-model', seed=-1, share_decoder_embeddings=False, start_checkpoint_at=0, start_decay_at=8, start_epoch=1, train_from='', train_from_state_dict='', truncated_decoder=0, warmup_steps=4000, word_vec_size=500)
WARNING: You have a CUDA device, should run with -gpus 0
Loading data from 'data/demo.train.pt'
 * vocabulary size. source = 24999; target = 35820
 * number of training sentences. 10000
 * maximum batch size. 64
Building model...
Intializing params
* number of parameters: 58121320
('generator.0.weight', 17910000)
('generator.0.bias', 35820)
('encoder: ', 16507500)
('decoder: ', 23668000)
Traceback (most recent call last):
  File "train.py", line 465, in <module>
    main()
  File "train.py", line 461, in main
    trainModel(model, trainData, validData, dataset, optim)
  File "train.py", line 220, in trainModel
    os.mkdir(model_dirname)
OSError: [Errno 2] No such file or directory: ''

I googled for a long time, but never succeed. Who can help me solve the problems please? Thank you very much!!

**Comment by bpopeters** [comment]
Hi, I had a look at this and there appears to be an error in how train.py handles the `-save_model` path when it doesn't include a directory. For a quick work-around, maybe you could try making a directory with a name like "foo" and running

`python train.py -data data/demo.train.pt -save_model foo/demo-model`

I'll see if I can fix the bug.

**Comment by bpopeters** [comment]
Incidentally, since you're running it on a machine with a GPU, you might want to add `-gpu 0` as well.

**Comment by VictorLi2017** [comment]
@bpopeters Thanks for your kind advice. Yesterday I updated the latest version of Pytorch, when I ran the same command, it come the errors and broke down. Seems quite terrible.
OpenNMT-py/onmt/Loss.py:124: UserWarning: self and other not broadcastable, but have the same number of elements.  Falling back to deprecated pointwise behavior.
  num_correct_t = pred_t.eq(targ_t.data) \
OpenNMT-py/onmt/Loss.py:125: UserWarning: self and mask not broadcastable, but have the same number of elements.  Falling back to deprecated pointwise behavior.
  .masked_select(non_padding) \
Write failed: Broken pipe
